vta/hardware/chisel/src/main/scala/core/Fetch.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package vta.core
+
+import chisel3._
+import chisel3.util._
+import vta.util.config._
+import vta.shell._
+
+class Fetch(debug: Boolean = false)(implicit p: Parameters) extends Module {
```

Comment:
It might be worth putting a single doc comment on each module which describes the high-level behavior of it. I know it might be a lot of work now, but will act as a roadmap for future devs.

Reply:
I just added some of this feedback [here](https://github.com/vegaluisjose/tvm/commit/cad584c16a687dbe092b2c9116f2d38abad36793). Still working on the remaining modules.